src/Areas/Server/Commands/ToolLoading/RegistryToolLoader.cs

```diff
@@ -118,35 +118,18 @@ public override async ValueTask<CallToolResult> CallToolHandler(RequestContext<C
 
     /// <summary>
     /// Transforms tool call arguments to a parameters dictionary.
+    /// This transformation is used because McpClientExtensions.CallToolAsync expects parameters as Dictionary&lt;string, object?&gt;.
     /// </summary>
     /// <param name="args">The arguments to transform to parameters.</param>
-    /// <returns>A dictionary of parameter names and values.</returns>
+    /// <returns>A dictionary of parameter names and values compatible with McpClientExtensions.CallToolAsync.</returns>
     private static Dictionary<string, object?> TransformArgumentsToDictionary(IReadOnlyDictionary<string, JsonElement>? args)
     {
         if (args == null)
         {
             return [];
         }
 
-        var parameters = new Dictionary<string, object?>();
-        foreach (var kvp in args)
-        {
-            // For simple types, extract the value directly
-            // For complex types, keep as JsonElement (which MCP client can handle)
-            parameters[kvp.Key] = kvp.Value.ValueKind switch
-            {
-                JsonValueKind.String => kvp.Value.GetString(),
-                JsonValueKind.Number when kvp.Value.TryGetInt32(out var intValue) => intValue,
-                JsonValueKind.Number when kvp.Value.TryGetInt64(out var longValue) => longValue,
-                JsonValueKind.Number when kvp.Value.TryGetDouble(out var doubleValue) => doubleValue,
-                JsonValueKind.True => true,
-                JsonValueKind.False => false,
-                JsonValueKind.Null => null,
```

Comment:
Even if we do these conversions, the C# MCP SDK will still turn/serialize everything into a `JsonElement` instance, those are identical to the `JsonElement` represented by `kvp.Value`.  Here is the debug view from MCP SDK after its primitive -> JsonElement serialization.

<img width="963" height="541" alt="Image" src="https://github.com/user-attachments/assets/ae86c9eb-3125-428f-9398-6ef21611810d" />

The MCP SDK has a fast path (noop) for JsonElement which we can take advantage here.

Reply:
@wbreza, the callsite [here](https://github.com/anuchandy/azure-mcp/blob/6eec2e1801febce820937512f55a4aadf5823fab/src/Areas/Server/Commands/ToolLoading/RegistryToolLoader.cs#L115) appear as

```C#
var parameters = TransformArgumentsToDictionary(request.Params.Arguments);
return await mcpClient.CallToolAsync(request.Params.Name, parameters, cancellationToken: cancellationToken);
```

but if we look at in all other loaders, we specifically extract tool parameters from "request.Params.Arguments["parameters"]" and pass it to "mcpClient.CallToolAsync". But only in case of `RegistryToolLoader`, we pass the whole "request.Params.Arguments" to "mcpClient.CallToolAsync", I didn't change this behavior but wanted to check with you if this was intentional.